http/cookies.lua

```diff
@@ -0,0 +1,324 @@
+local http_patts = require "lpeg_patterns.http"
+local util = require "http.util"
+local psl = require "psl"
+
+local function parse_set_cookie(text_cookie, request, time)
+	local domain = request.headers:get("host")
+	local key, value, matched_cookie = assert(http_patts.Set_Cookie:match(
+		text_cookie))
+	-- allow custom time for specs to not error
+	time = time or os.time()
+	local cookie = {
+		creation = time;
+		last_access = time;
+		persistent =
+			not not (matched_cookie.expires or matched_cookie["max-age"]);
+		domain = matched_cookie.domain or domain;
+		path = matched_cookie.path or request.headers:get(":path");
+		secure = matched_cookie.secure or false;
+		http_only = matched_cookie.httponly or false;
+		key = key;
+		value = value;
+		host_only = not not matched_cookie.domain
+	}
+	cookie.indexname = cookie.domain .."|".. cookie.path .."|".. cookie.key
```

Comment:
Changed to a function and changed the separator to `\1`. However, I don't see much of a use for it and might remove it if I don't see anything that uses it.